main/src/library/LazyList.flix

```diff
@@ -955,4 +955,29 @@ namespace LazyList {
         case LList(xs)    => partitionAcc(f, force xs, k)
     }
 
+    ///
+    /// `l1` is the longest prefix of `l` that satisfies the predicate `f`.
+    /// `l2` is the remainder of `l`.
+    ///
+    pub def span(f: a -> Bool & ef, l: LazyList[a]): (LazyList[a], LazyList[a]) & ef =
```

Comment:
Does this force the entire list? I would think perhaps I could be lazy in the second list?

Reply:
Second, if `f` is pure, could this be even more lazy or is that not really practical?

Reply:
Oh right, good point. I'll have a look at it and get back to you. Marking this PR as a draft for now.

Reply:
I guess you could have something to to the effect of returning `(LList(lazy takeWhile(f, l)), LList(lazy dropWhile(f, l)))`